lib/src/main/java/com/diffplug/spotless/java/ImportSorterImpl.java

```diff
@@ -25,12 +27,41 @@
 // which itself is licensed under the Apache 2.0 license.
 final class ImportSorterImpl {
 
-	private final List<String> template = new ArrayList<>();
+	private static final String CATCH_ALL_SUBGROUP = "";
+	private static final String STATIC_KEYWORD = "static ";
+	private static final String STATIC_SYMBOL = "\\#";
+	private static final String SUBGROUP_SEPARATOR = "|";
+
+	private final List<ImportsGroup> importsGroups;
 	private final Map<String, List<String>> matchingImports = new HashMap<>();
 	private final List<String> notMatching = new ArrayList<>();
 	private final Set<String> allImportOrderItems = new HashSet<>();
 	private final Comparator<String> ordering;
 
+	// An ImportsGroup is a group of imports ; each group is separated by blank lines.
+	// A group is composed of subgroups : imports are sorted by subgroup.
+	private static class ImportsGroup {
+
+		private final List<String> subGroups;
+
+		public ImportsGroup(String importOrder) {
+			this.subGroups = Stream.of(importOrder.split("\\" + SUBGROUP_SEPARATOR))
```

Comment:
This line has a bug that only `|\\#` can be split as `{ "", "\\#" }`, but `\\#|` will be split as `{ "\\#" }` so configure `static` subgroup before catch all subgroup is impossible.

Reply:
cc @murdos @nedtwigg 

Also, it seems that the doc can be incorrect since the correct syntax is `\#` instead of `\\#`.

Reply:
```suggestion
			this.subGroups = Stream.of(importOrder.split("\\" + SUBGROUP_SEPARATOR, -1))
```

... can help.